### ISSUE TITLE
Offline export

### ISSUE THREAD

**Issue by aurora-potter** [wallabag/android-app#606]
using an online service is evil idea

**Comment by di72nn** [comment]
Well, the app is useless anyway unless it can connect to this "online service" once in a while, so it is a fair trade-off.

I guess, another viable compromise is to allow to export the article as HTML (as it is stored locally), so some external app can convert it to whatever format user wants.

I haven't looked into this, but including libs to locally convert content to PDF will bloat the app.

**Comment by aurora-potter** [comment]
Providing an export option is nice.
I recommend html & txt by default and pdf by a third-party plugin

**Comment by Strubbl** [comment]
Why a third party plugin? Which one? There is already some 3rdparty plugin server-side.
Why should the app take over the task of what wallabag server does?

**Comment by aurora-potter** [comment]
@Strubbl
Is the PDF conversion server the same as wallabag?

But you could just have an offline PDF conversion option.
You don't have to make the library yourself. there are lots of html to pdf libraries out there.

**Comment by di72nn** [comment]
Yeah, the app uses only wallabag server for PDF (and other formats) conversion.

The feature to download articles in various formats was available on the server for a long time. Then somebody wanted to be able to easily download the articles via the app, so that "download PDF" feature was added to the app. That feature didn't take much work and didn't bloat the app.

Client-side conversion, on the other hand, wasn't demanded so far and would result in some negative effects (increased app size, probable dependency on some native libs). So there's no plans to implement it.

HTML export should be easy to implement. HTML can be converted to other formats **using other apps if somebody needs it**.

**Comment by aurora-potter** [comment]
@di72nn
Offline html export is nice.
